packages/explorerkit-server/src/components/idls.ts

```diff
@@ -53,21 +84,85 @@ const getInMemoryProgramIdl = (programId: string): IdlItem | null => {
   return IN_MEMORY_PROGRAM_IDLS.get(programId) || null;
 };
 
-const deserializeIdl = (idl: string): IdlItem | null => {
-  try {
-    const item = JSON.parse(idl) as MaybeIdl;
+const idlRefreshQueue = new Set<string>();
+
+export function addIdlToRefreshQueue(programId: string) {
+  idlRefreshQueue.add(programId);
+}
+
+const BATCH_SIZE = 10;
+
+export async function refreshIdlsInQueue() {
+  const programIds = Array.from(idlRefreshQueue.values());
+  const programIdsChunks = chunkArray(programIds, BATCH_SIZE);
+
+  const cache = getSharedDep("cache");
 
-    if (item.type === "MISSING") {
-      return null;
+  for (const programIds of programIdsChunks) {
+    const idls = await getMultipleProgramIdls(programIds).catch((error) => {
+      console.error("Error fetching idls", error);
+      return [] as IdlItem[];
+    });
+
+    const idlsByProgramId = idls.reduce((acc, idl) => {
+      if (idl) {
+        acc.set(idl.programId, idl);
+      }
+      return acc;
+    }, new Map<string, IdlItem>());
+
+    for (const programId of programIds) {
+      const idl = idlsByProgramId.get(programId) ?? null;
+      const serializedIdl = serializeIdl(idl, new Date(Date.now() + IDL_STALE_TIME * 1000));
+
+      await cache.set(programId, serializedIdl, IDL_CACHE_TTL).catch((error) => {
+        console.error("Error caching idl", error);
+      });
+
+      idlRefreshQueue.delete(programId);
+    }
+  }
+}
+
+export function initIdlsRefreshBackgroundJob(idlRefreshIntervalMs: number) {
+  const queueSizeMetric = new Gauge({
+    name: "idls_refresh_queue_size",
+    help: "Number of idls to refresh",
+    registers: [register],
+  });
+
```

Comment:
NIT: Could we track the refresh timings as well? Just so we know if it starts getting slower and slower